horace_core/sqw/@sqw/resolution_plot.m

```diff
@@ -190,42 +205,57 @@
 [inst, all_inst] = get_inst_class(w);
 if isempty(inst)
     if all_inst
-        error('The instrument type is not the same for all contributing raw data files')
+        error('HORACE:resolution_plot:invalid_argument',...
+            'The instrument type is not the same for all contributing raw data files')
     else
-        error('The instrument has not been defined for all contributing raw data files')
+        error('HORACE:resolution_plot:invalid_argument',...
+            'The instrument has not been defined for all contributing raw data files')
     end
 end
 if strcmp(inst,'IX_inst_DGfermi')
     resfun_model = @tobyfit_DGfermi_resfun_covariance;
 elseif strcmp(inst,'IX_inst_DGdisk')
     resfun_model = @tobyfit_DGdisk_resfun_covariance;
 else
-    error('No resolution function model implemented for this instrument')
+    error('HORACE:resolution_plot:invalid_argument',...
+        'No resolution function model implemented for this instrument')
 end
 
-iax_plot = [w.data.pax, iax];
-if npixtot==0
-    % Special case of no data, one header, one detector
-    covariance_matrix = resfun_model(w);
-    resolution_plot_private ([0,0], covariance_matrix, iax_plot, false)
+% Compute resolution function, and plot unless 'noplot' requested
+[kept, ipix] = get_nearest_pixels (w, xp);
+if numel(ipix)>0
+    [cov_proj, cov_spec, cov_hkle] = resfun_model(w, ipix);
+    xp_kept = xp(kept,:);
+    iax_plot = [w.data.pax, iax];
+    if ~present.noplot
+        for i=1:numel(ipix)
+            resolution_plot_private (xp_kept(i,:), cov_proj(:,:,i),...
+                iax_plot, flipped_display_axes)
+        end
+    end
 else
-    [xp_ok, ipix] = get_nearest_pixels (w, xp);
-    covariance_matrix = resfun_model(w, ipix);
-    for i=1:numel(ipix)
-        resolution_plot_private (xp_ok(i,:), covariance_matrix(:,:,i),...
-            iax_plot, flip)
+    if ~present.noplot
+        error('HORACE:resolution_plot:invalid_argument',...
+            'Resolution ellipsoid(s) all lie outside bins with data')
+    else
+        cov_proj = zeros(4,4,0);
+        cov_spec = zeros(4,4,0);
+        cov_hkle = zeros(4,4,0);
     end
 end
 
-% If newplot, then rescale limits
-if newplot
-    lx('round')
-    ly('round')
-end
-
 
 % Return covariance matrix, if requested
 % --------------------------------------
-if nargout==1
-    varargout{1} = covariance_matrix;
+if nargout>=1
```

Comment:
Given you know the argument count, would it not be better to just make these static arguments? Matlab ignores non-returned arguments anyway.

Reply:
This is a little trick. If one has static return arguments, then typing at the matlab prompt
>> resolution_plot (...)
would result in the first return argument being printed at the screen unless the line was terminated with a semi-colon. Mostly I expect this will be used without the uyser expecting a return argument - it is a visualisation utility - but they are there in case they are wanted (e.g. as I have done in the past). The odd looking if nargin>=1 etc lines enable this to be implemented.